client/state/posts/utils.js

```diff
@@ -241,6 +243,33 @@ export function normalizeTermsForApi( post ) {
 }
 
 /**
+ * Returns truthy if local terms object is different than API response
+ *
+ * @param  {Object}  localTermEdits local state of term edits
+ * @param  {Object}  savedTerms     term object returned from API POST
+ * @return {Boolean}                are there differences in local edits vs saved terms
+ */
+export function hasTermEditDifferences( localTermEdits, savedTerms ) {
+	if ( ! localTermEdits || ! savedTerms ) {
+		return false;
```

Comment:
Would we expect this to return `true` if `hasTermEditDifferences( { post_tag: [ ... ] }, null )` ? Not sure anything like this would ever happen.